Parent/Parent/Dashboard/DashboardViewController.swift

```diff
@@ -152,45 +147,16 @@ class DashboardViewController: UIViewController, CustomNavbarProtocol {
 
     }
 
-    func setup() throws {
-        guard studentCollection == nil else { return }
-        viewState.isSiteAdmin = session.isSiteAdmin
-        studentCollection = try Student.observedStudentsCollection(session)
-        studentCountObserver = try Student.countOfObservedStudentsObserver(session) { [weak self] count in
-
-            // Check to see if all students were all removed during
-            // the current user session
-            var noMoreLinkedStudents = false
-            if count == 0,
-                let state = self?.viewState,
-                state.studentCount > 0,
-                state.isValidObserver {
-                noMoreLinkedStudents = true
-            }
-
-            self?.viewState.studentCount = count
-            self?.configureStudentMenu()
-
-            if (noMoreLinkedStudents) {
-                DispatchQueue.main.async {
-                    self?.updateMainView()
-                }
+    func updateStudents() {
+        if !students.pending {
+            if students.error != nil {
+                //  TODO: - may need to check the type of error
+                viewState.isValidObserver = false
+                return
```

Comment:
Do we need to call `configureStudentMenu` or `updateMainView` so that the viewState changes are displayed? Looks like the old code also handled `studentCount` being 0. We probably still want to handle that right?